tests/constrainedExample.h

```diff
@@ -32,44 +32,52 @@ using gtsam::Vector1;
 using gtsam::Vector2;
 using gtsam::Vector2_;
 
-/** First cost function. */
-double cost1(const double &x1, const double &x2,
-             gtsam::OptionalJacobian<1, 1> H_x1 = boost::none,
-             gtsam::OptionalJacobian<1, 1> H_x2 = boost::none) {
-  double result = x1 + exp(-x2);
-  if (H_x1) H_x1->setConstant(1.0);
-  if (H_x2) H_x2->setConstant(-exp(-x2));
+/// Exponential function e^x.
+double exp_func(const double& x,
+                gtsam::OptionalJacobian<1, 1> H1 = boost::none) {
+  double result = exp(x);
+  if (H1) H1->setConstant(result);
   return result;
 }
 
-/** Second cost function. */
-double cost2(const double &x1, const double &x2,
-             gtsam::OptionalJacobian<1, 1> H_x1 = boost::none,
-             gtsam::OptionalJacobian<1, 1> H_x2 = boost::none) {
-  double result = x1 * x1 + 2 * x2 + 1;
-  if (H_x1) H_x1->setConstant(2 * x1);
-  if (H_x2) H_x2->setConstant(2.0);
-  return result;
-}
+/// Exponential expression e^x.
+Double_ exp(const Double_& x) { return Double_(exp_func, x); }
 
-/** Constraint function g(x). */
-double constraint1(const double &x1, const double &x2,
-                   gtsam::OptionalJacobian<1, 1> H_x1 = boost::none,
-                   gtsam::OptionalJacobian<1, 1> H_x2 = boost::none) {
-  double result = x1 + x1 * x1 * x1 + x2 + x2 * x2;
-  if (H_x1) H_x1->setConstant(1 + 3 * x1 * x1);
-  if (H_x2) H_x2->setConstant(1 + 2 * x2);
-  return result;
+/// Pow functor used for pow function.
+class PowFunctor {
+ private:
+  double c_;
+
+ public:
```

Comment:
Cool! Should c not be an int, though?

Reply:
we are only using int in the example, but we can make it double to be general.